[Keyboard] add Stoutgat v2; move Stoutgat v1

vattern opened the pull request:
<!--- Provide a general summary of your changes in the title above. -->

<!--- This template is entirely optional and can be removed, but is here to help both you and us. -->
<!--- Anything on lines wrapped in comments like these will not show up in the final text. -->

## Description

<!--- Describe your changes in detail here. -->

## Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply. -->
- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

drashna commented on keyboards/tkw/stoutgatv2/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,77 @@
+#include QMK_KEYBOARD_H
+
+#define _BL 0
+#define _FL 1
```
```suggestion
enun layers {
    _BL,
    _FL,
};
```

drashna commented on keyboards/tkw/stoutgatv2/rules.mk:
```diff
@@ -0,0 +1,26 @@
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
```
```suggestion
```

drashna reviewed (changes requested):
Additionally, it looks like there are changes to the V-USB folder.  These need to be reverted.

fauxpark commented on keyboards/tkw/stoutgatv2/config.h:
```diff
@@ -0,0 +1,69 @@
+#define PRODUCT_ID 0x7811
+#define DEVICE_VER 0x0002
+#define MANUFACTURER TKW
+#define PRODUCT Stoutgat 65% with RGB and dual encoders
```
```suggestion
#define PRODUCT Stoutgat 65%
```

fauxpark commented on keyboards/tkw/stoutgatv2/rules.mk:
```diff
@@ -0,0 +1,26 @@
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+ENCODER_ENABLE = yes        # Enable rotary encoder support
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
```
```suggestion
BLUETOOTH_ENABLE = no       # Enable Bluetooth
```

noroadsleft reviewed (commented):
Any relation here to [`keyboards/stoutgat`](https://github.com/qmk/qmk_firmware/tree/3d4f0028d60cebc829ac9c947d1a61cc840d05c0/keyboards/stoutgat)? If this is a new version of that board, you could add this one as a subfolder there (or move that one here).

vattern commented:
They are related, however I would prefer to move keyboards/stoutgat to
keyboards/tkw/stoutgat if possible.

On Thu, Sep 10, 2020 at 7:39 PM James Young <noreply@vektorfoto.co.za>
wrote:

> *@noroadsleft* commented on this pull request.
>
> Any relation here to keyboards/stoutgat? If this is a new version of that
> board, you could add this one as a subfolder there.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/qmk/qmk_firmware/pull/10233#pullrequestreview-486212085>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABDWJ27LRLOTEMTBVSHW6BTSFEML7ANCNFSM4QYGDJ2Q>
> .
>

noroadsleft commented:
```shell
cd keyboards/

# move the stoutgat v1:
git mv stoutgat/ tkw/stoutgat/v1/
cd tkw/stoutgat/v1/
git mv stoutgat/ ../
cd ../
git mv stoutgat/ v1/

# rework the stoutgat v2:
cd ../
git mv stoutgatv2/ stoutgat/v2/

# fix the filenames
git mv v1/stoutgat.c v1/v1.c
git mv v1/stoutgat.h v1/v1.h
git mv v2/stoutgatv2.c v2/v2.c
git mv v2/stoutgatv2.h v2/v2.h
```

Then fix the includes in the `v1.c` and `v2.c` files.

Also, if the v2 only supports the f411, you can move/merge all those files with the equivalent files in `v2/` as appropriate.

vattern commented:
I have now moved v1 and v2 to under tkw/stoutgat, and updated as needed.

On Fri, Sep 11, 2020 at 10:57 PM James Young <noreply@vektorfoto.co.za>
wrote:

> cd keyboards/
> # move the stoutgat v1:
> git mv stoutgat/ tkw/stoutgat/v1/cd tkw/stoutgat/v1/
> git mv stoutgat/ ../cd ../
> git mv stoutgat/ v1/
> # rework the stoutgat v2:cd ../
> git mv stoutgatv2/ stoutgat/v2/
> # fix the filenames
> git mv v1/stoutgat.c v1/v1.c
> git mv v1/stoutgat.h v1/v1.h
> git mv v2/stoutgatv2.c v2/v2.c
> git mv v2/stoutgatv2.h v2/v2.h
>
> Then fix the includes in the v1.c and v2.c files.
>
> Also, if the v2 only supports the f411, you can move/merge all those files
> with the equivalent files in v2/ as appropriate.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/qmk/qmk_firmware/pull/10233#issuecomment-691329491>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABDWJ2ZLFLORTOGQ3MTJSQ3SFKMNDANCNFSM4QYGDJ2Q>
> .
>

noroadsleft commented on keyboards/tkw/stoutgat/v2/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,79 @@
+
+   const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+     [0] = LAYOUT_encoder(
+       KC_MPLY,KC_ESC,
```
Correct me if I'm wrong, but:

- This keymap is copied and modified from the V1 PCB.
- Looking at the schematics from your GitHub, `KC_ESC` here seems like it wouldn't actually work:
  - The macro assigns this position to `K48`, which exists on the V1 but not on the V2.
  - The V2 seems to support three rotary encoders:
    - `SW18`: Ctrl + Right or Left on the Base layer when turned, in the place that is normally the Escape key. `KC_GESC` when tapped.
    - `SW81`: Also Ctrl + Right or Left on the Base layer when turned (same encoder pins as `SW18`), located next to the left Shift key. `KC_MPLY` when tapped.
    - `SW12`: Page Down or Up when turned, `KC_GRV` when tapped. Located in the top right corner of the board.

vattern replied:
I've removed the K48 from the macro, thanks for spotting that.
The idea is that you either mount an encoder at SW18 OR SW81, but not both.

noroadsleft commented on keyboards/tkw/stoutgat/v2/v2.h:
```diff
@@ -0,0 +1,76 @@
+#include "quantum.h"
+
+#define LAYOUT_encoder( \
+       K44, \
+  K00, K01, K02, K03, K04, K05, K06, K07, K08, K09, K0A, K0B, K0C, K0D, K0E, \
+  K10, K11, K12, K13, K14, K15, K16, K17, K18, K19, K1A, K1B, K1C,      K1E, \
+  K20, K21, K22, K23, K24, K25, K26, K27, K28, K29, K2A, K2B, K2D, K1D, K2E, \
+  K30, K31, K32, K33, K34, K35, K36, K37, K38, K39, K3A, K3B, K3C, K3D, K3E, \
+  K40, K41, K42,                K46,           K49, K4A, K4B, K4C, K4D, K4E  \
```
This section should be formatted to resemble the assembled hardware, like so:

```suggestion
          K00, K01, K02, K03, K04, K05, K06, K07, K08, K09, K0A, K0B, K0C, K0D, K0E, \
          K10, K11, K12, K13, K14, K15, K16, K17, K18, K19, K1A, K1B, K1C,      K1E, \
          K20, K21, K22, K23, K24, K25, K26, K27, K28, K29, K2A, K2B, K2D, K1D, K2E, \
    K44,  K30, K31, K32, K33, K34, K35, K36, K37, K38, K39, K3A, K3B, K3C, K3D, K3E, \
          K40, K41, K42,                K46,           K49, K4A, K4B, K4C, K4D, K4E  \
```

noroadsleft commented on keyboards/tkw/stoutgat/v2/keymaps/default/keymap.c:
```diff
@@ -0,0 +1,79 @@
+
+
+
+   const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+     [0] = LAYOUT_encoder(
+       KC_MPLY,
+         KC_GESC, KC_1,    KC_2,    KC_3,    KC_4,   KC_5,   KC_6,   KC_7,   KC_8,   KC_9,    KC_0,    KC_MINS,  KC_EQL,  KC_BSPC,  KC_GRV,
+         KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,   KC_T,   KC_Y,   KC_U,   KC_I,   KC_O,    KC_P,    KC_LBRC,  KC_RBRC,           KC_DEL,
+         KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,   KC_G,   KC_H,   KC_J,   KC_K,   KC_L,    KC_SCLN, KC_QUOT,  KC_NUHS, KC_ENT,   KC_PGUP,
+         KC_LSFT, KC_NUBS, KC_Z,    KC_X,    KC_C,   KC_V,   KC_B,   KC_N,   KC_M,   KC_COMM, KC_DOT,  KC_SLSH,  KC_RSFT, KC_UP,    KC_PGDN,
+         KC_LCTL, KC_LGUI, KC_LALT,                         KC_SPC ,KC_RALT, MO(1),  KC_RCTL,  KC_LEFT,  KC_DOWN,  KC_RIGHT),
+
+     [1] = LAYOUT_encoder(
+       KC_MPLY,
+        /* esc      1        2        3        4       5       6       7       8       9         0        -         =     bkspc       `~  */
+         KC_GRV,  KC_F1,   KC_F2,   KC_F3,   KC_F4,  KC_F5,  KC_F6,  KC_F7,  KC_F8,  KC_F9,   KC_F10,  KC_F11,   KC_F12,  KC_DEL,   RESET,
+        /*  tab      Q       W        E        R        T      Y        U      I        O        P        [         ]               delete*/
+         RGB_MOD, RGB_VAI, RGB_SAI, RGB_HUI, KC_TRNS,  KC_TRNS,KC_TRNS,KC_TRNS,KC_INS, KC_TRNS, KC_TRNS, KC_TRNS,  KC_PAUS,           KC_TRNS,
+        /*  caps     A       S        D        F        G      H        J      K        L        ;        '          #     enter     pg up*/
+         RGB_TOG, RGB_VAD, RGB_SAD, RGB_HUD, KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS, KC_TRNS, KC_TRNS,  KC_INS,  KC_TRNS,  KC_HOME,
+        /* shift     \       Z         X        C       V       B       N      M        ,        .        /        shift     up      pg dn*/
+         KC_LSFT, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,KC_TRNS,KC_TRNS,KC_TRNS,KC_MUTE, KC_TRNS, KC_TRNS, KC_TRNS,  KC_RSFT, KC_VOLU,  KC_END,
+        /* ctrl     win      alt                              space                     alt      fn       ctrl      left     down     right*/
+         KC_LCTL, KC_LGUI, KC_LALT,                          KC_TRNS,                 KC_RALT, KC_TRNS, KC_RCTL,  KC_TRNS, KC_VOLD,  KC_TRNS)
+   };
```
Goes with the change to `v2.h`:

```suggestion
const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
    [0] = LAYOUT_encoder(
                 KC_GESC, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_GRV,  
                 KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC,          KC_DEL,  
                 KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT, KC_NUHS, KC_ENT,  KC_PGUP, 
        KC_MPLY, KC_LSFT, KC_NUBS, KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT, KC_UP,   KC_PGDN, 
                 KC_LCTL, KC_LGUI, KC_LALT,                            KC_SPC,                    KC_RALT, MO(1),   KC_RCTL, KC_LEFT, KC_DOWN, KC_RGHT     
    ),

    [1] = LAYOUT_encoder(
                 /* esc   1        2        3        4        5        6        7        8        9        0        -        =        bkspc    `~      */
                 KC_GRV,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  KC_DEL,  RESET, 
                 /* tab   Q        W        E        R        T        Y        U        I        O        P        [        ]                 delete  */
                 RGB_MOD, RGB_VAI, RGB_SAI, RGB_HUI, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_INS,  KC_TRNS, KC_TRNS, KC_TRNS, KC_PAUS,          KC_TRNS, 
                 /* caps  A        S        D        F        G        H        J        K        L        ;        '        #        enter    pg up   */
                 RGB_TOG, RGB_VAD, RGB_SAD, RGB_HUD, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_INS,  KC_TRNS, KC_HOME, 
                 /* shift \        Z        X        C        V        B        N        M        ,        .        /        shift    up       pg dn   */
        KC_MPLY, KC_LSFT, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_MUTE, KC_TRNS, KC_TRNS, KC_TRNS, KC_RSFT, KC_VOLU, KC_END, 
                 /* ctrl  win      alt                                 space                      alt      fn       ctrl     left     down     right   */
                 KC_LCTL, KC_LGUI, KC_LALT,                            KC_TRNS,                   KC_RALT, KC_TRNS, KC_RCTL, KC_TRNS, KC_VOLD, KC_TRNS
    )
};
```

noroadsleft commented on keyboards/tkw/stoutgat/v2/info.json:
```diff
@@ -0,0 +1,92 @@
+    "url": "https://github.com/vattern/stoutgatv2",
+    "maintainer": "vattern",
+    "manufacturer": "tkw",
+    "width": 17,
+    "height": 6,
```
```suggestion
    "width": 17.25,
    "height": 5,
```

noroadsleft commented on keyboards/tkw/stoutgat/v2/info.json:
```diff
@@ -0,0 +1,92 @@
+                {"x": 4.75, "y": 3},
+                {"x": 5.75, "y": 3},
+                {"x": 6.75, "y": 3},
+                {"x": 7.75, "y": 3},
+                {"x": 8.75, "y": 3},
+                {"x": 9.75, "y": 3},
+                {"x": 10.75, "y": 3},
+                {"x": 11.75, "y": 3},
+                {"x": 12.75, "y": 3},
+                {"x": 13.75, "y": 3},
+                {"x": 14.75, "y": 2, "w": 1.25, "h": 2},
+                {"x": 16, "y": 3},
+
+                {"x": 1, "y": 4, "w": 1.25},
+                {"x": 2.25, "y": 4},
+                {"x": 3.25, "y": 4},
+                {"x": 4.25, "y": 4},
+                {"x": 5.25, "y": 4},
+                {"x": 6.25, "y": 4},
+                {"x": 7.25, "y": 4},
+                {"x": 8.25, "y": 4},
+                {"x": 9.25, "y": 4},
+                {"x": 10.25, "y": 4},
+                {"x": 11.25, "y": 4},
+                {"x": 12.25, "y": 4},
+                {"x": 13.25, "y": 4, "w": 1.75},
+                {"x": 15, "y": 4},
+                {"x": 16, "y": 4},
+
+                {"x": 1, "y": 5, "w": 1.25},
+                {"x": 2.25, "y": 5, "w": 1.25},
+                {"x": 3.5, "y": 5, "w": 1.25},
+                {"x": 4.75, "y": 5, "w": 6.25},
+
+                {"x": 11, "y": 5},
+                {"x": 12, "y": 5},
+                {"x": 13, "y": 5},
+                {"x": 14, "y": 5},
+                {"x": 15, "y": 5},
+                {"x": 16, "y": 5}
```
And rearranging here too.

```suggestion
                {"x": 1.25, "y": 0},
                {"x": 2.25, "y": 0},
                {"x": 3.25, "y": 0},
                {"x": 4.25, "y": 0},
                {"x": 5.25, "y": 0},
                {"x": 6.25, "y": 0},
                {"x": 7.25, "y": 0},
                {"x": 8.25, "y": 0},
                {"x": 9.25, "y": 0},
                {"x": 10.25, "y": 0},
                {"x": 11.25, "y": 0},
                {"x": 12.25, "y": 0},
                {"x": 13.25, "y": 0},
                {"x": 14.25, "y": 0, "w": 2},
                {"x": 16.25, "y": 0},

                {"x": 1.25, "y": 1, "w": 1.5},
                {"x": 2.75, "y": 1},
                {"x": 3.75, "y": 1},
                {"x": 4.75, "y": 1},
                {"x": 5.75, "y": 1},
                {"x": 6.75, "y": 1},
                {"x": 7.75, "y": 1},
                {"x": 8.75, "y": 1},
                {"x": 9.75, "y": 1},
                {"x": 10.75, "y": 1},
                {"x": 11.75, "y": 1},
                {"x": 12.75, "y": 1},
                {"x": 13.75, "y": 1},
                {"x": 16.25, "y": 1},

                {"x": 1.25, "y": 2, "w": 1.75},
                {"x": 3.0, "y": 2},
                {"x": 4.0, "y": 2},
                {"x": 5.0, "y": 2},
                {"x": 6.0, "y": 2},
                {"x": 7.0, "y": 2},
                {"x": 8.0, "y": 2},
                {"x": 9.0, "y": 2},
                {"x": 10.0, "y": 2},
                {"x": 11.0, "y": 2},
                {"x": 12.0, "y": 2},
                {"x": 13.0, "y": 2},
                {"x": 14.0, "y": 2},
                {"x": 15.0, "y": 1, "w": 1.25, "h": 2},
                {"x": 16.25, "y": 2},

                {"x": 0, "y": 3},

                {"x": 1.25, "y": 3, "w": 1.25},
                {"x": 2.5, "y": 3},
                {"x": 3.5, "y": 3},
                {"x": 4.5, "y": 3},
                {"x": 5.5, "y": 3},
                {"x": 6.5, "y": 3},
                {"x": 7.5, "y": 3},
                {"x": 8.5, "y": 3},
                {"x": 9.5, "y": 3},
                {"x": 10.5, "y": 3},
                {"x": 11.5, "y": 3},
                {"x": 12.5, "y": 3},
                {"x": 13.5, "y": 3, "w": 1.75},
                {"x": 15.25, "y": 3},
                {"x": 16.25, "y": 3},

                {"x": 1.25, "y": 4, "w": 1.25},
                {"x": 2.5, "y": 4, "w": 1.25},
                {"x": 3.75, "y": 4, "w": 1.25},
                {"x": 5.0, "y": 4, "w": 6.25},
                {"x": 11.25, "y": 4},
                {"x": 12.25, "y": 4},
                {"x": 13.25, "y": 4},
                {"x": 14.25, "y": 4},
                {"x": 15.25, "y": 4},
                {"x": 16.25, "y": 4}
```

noroadsleft commented on keyboards/tkw/stoutgat/v2/keymaps/qmk/keymap.json:
```diff
@@ -0,0 +1,158 @@
+{
```
If you're intending for this to be your personal keymap, the folder should be `keyboards/tkw/stoutgat/v2/keymaps/thys/`.

If you're intending for it to be the default for QMK Configurator:

1. The file to achieve that doesn't go in this repository.
2. It would need to be updated due to the change to `LAYOUT_encoder`.
3. It doesn't match the assignment used in the `default` keymap, which is preferred.

vattern replied:
removed

noroadsleft commented:
Thanks!
